uplink/ecclient/client.go

```diff
@@ -213,42 +213,44 @@ func (ec *ecClient) putPiece(ctx, parent context.Context, limit *pb.AddressedOrd
 		)
 		return nil, err
 	}
+
 	defer func() {
-		if ctx.Err() != nil || err != nil {
-			hash = nil
+		if err != nil {
 			err = errs.Combine(err, upload.Cancel(ctx))
 			return
 		}
-		h, closeErr := upload.Commit(ctx)
-		hash = h
-		err = errs.Combine(err, closeErr)
+
+		hash, err = upload.Commit(ctx)
 	}()
 
 	_, err = sync2.Copy(ctx, upload, data)
 	// Canceled context means the piece upload was interrupted by user or due
 	// to slow connection. No error logging for this case.
-	if ctx.Err() == context.Canceled {
-		if parent.Err() == context.Canceled {
-			ec.log.Info("Upload to node canceled by user", zap.String("NodeID", storageNodeID.String()))
+	if err != nil {
+		if errs2.IsCanceled(err) {
+			if parent.Err() == context.Canceled {
+				ec.log.Info("Upload to node canceled by user", zap.String("NodeID", storageNodeID.String()))
+			} else {
+				ec.log.Debug("Node cut from upload due to slow connection", zap.String("NodeID", storageNodeID.String()))
+			}
 		} else {
-			ec.log.Debug("Node cut from upload due to slow connection", zap.String("NodeID", storageNodeID.String()))
-		}
-		err = context.Canceled
-	} else if err != nil {
-		nodeAddress := "nil"
-		if limit.GetStorageNodeAddress() != nil {
-			nodeAddress = limit.GetStorageNodeAddress().GetAddress()
+			nodeAddress := ""
+			if limit.GetStorageNodeAddress() != nil {
+				nodeAddress = limit.GetStorageNodeAddress().GetAddress()
+			}
+
+			ec.log.Debug("Failed uploading piece to node",
+				zap.String("PieceID", pieceID.String()),
```

Comment:
I saw that in this PR, @littleskunk mentioned that we should use `zap.Stringer` wherever possible. https://github.com/storj/storj/pull/3285#discussion_r335236248

Reply:
Yes, that's a good catch.
It's something new that I learnt the last week from a review of another of my PRs, however, here I didn't even see it as they were dragged from the current implementation.

I'm going to change them!

Reply:
Done